https://bugs.gentoo.org/910570
https://github.com/systemd/systemd/issues/28520
https://github.com/systemd/systemd/commit/eff91e2f3863f9e176b383e5c54741c64ca7a636

From eff91e2f3863f9e176b383e5c54741c64ca7a636 Mon Sep 17 00:00:00 2001
From: Luca Boccassi <bluca@debian.org>
Date: Wed, 26 Jul 2023 11:29:57 +0100
Subject: [PATCH] efi: link with -z nopack-relative-relocs

elf2efi.py cannot handle DT_RELR relocations, so disable it
if we can

Fixes https://github.com/systemd/systemd/issues/28520
--- a/src/boot/efi/meson.build
+++ b/src/boot/efi/meson.build
@@ -174,6 +174,10 @@ efi_c_ld_args = [
         '-T' + elf2efi_lds,
 ]
 
+# On CentOS 8 the nopack-relative-relocs linker flag is not supported, and we get:
+#   /usr/bin/ld.bfd: warning: -z nopack-relative-relocs ignored
+efi_c_ld_args += cc.get_supported_link_arguments('-Wl,-z,nopack-relative-relocs')
+
 # efi_c_args is explicitly passed to targets so that they can override distro-provided flags
 # that should not be used for EFI binaries.
 efi_disabled_c_args = cc.get_supported_arguments(
